klippa/src/main.rs

```diff
@@ -100,6 +121,34 @@ fn populate_gids(gid_str: &str) -> Result<BTreeSet<GlyphId>, InvalidInputError>
             result.insert(GlyphId::new(glyph_id));
         }
     }
-    result.insert(GlyphId::new(0_u16));
+    Ok(result)
+}
+
+fn parse_unicodes(unicode_str: &Option<String>) -> Result<IntSet<u32>, InvalidInputError> {
+    let mut result = IntSet::empty();
+    if unicode_str.is_none() {
+        return Ok(result);
+    }
+    let re = regex::Regex::new(r"[<+->{},;&#\\xXuUnNiI\n\t\v\f\r]").unwrap();
+    let s = re.replace_all(unicode_str.as_deref().unwrap(), " ");
+    for cp in s.split_whitespace() {
+        if let Some((start, end)) = cp.split_once('-') {
+            let start: u32 = start
+                .parse::<u32>()
+                .map_err(|_| InvalidInputError::InvalidUnicode(start.to_owned().clone()))?;
+            let end: u32 = end
+                .parse::<u32>()
+                .map_err(|_| InvalidInputError::InvalidUnicode(end.to_owned().clone()))?;
+            if start > end {
+                return Err(InvalidInputError::InvalidUnicodeRange { start, end });
+            }
+            result.extend(start..=end);
+        } else {
+            let unicode: u32 = cp
+                .parse::<u32>()
+                .map_err(|_| InvalidInputError::InvalidUnicode(cp.to_owned().clone()))?;
+            result.insert(unicode);
+        }
+    }
```

Comment:
do we care if the input is a valid `u32` but not a valid codepoint? Do we handle that later? it might make sense to also do that validation here..

Reply:
yeah we can probably skip here cause we will handle that later during planning phase